concepts/tuples/about.md

```diff
@@ -0,0 +1,119 @@
+# Tuples
+
+A [tuple][tuple] is a finite ordered list of elements which is immutable.
+Tuples requires all positions to have a fixed type this in turns means that the compiler knows what type each position is.
+The types used in a tuple can be different, but the types must be known at compile time.
+
+## Creating a Tuple
+
+Depending on if the tuples values types can be interprited under compilation, the tuple can be created in different ways.
+It is also important that the types of the values match the types specified in the tuple and that the number of values match the number of types specified.
+If the values are known at compile time, the tuple can be created using the tuple literal syntax.
+
+```crystal
+tuple = {1, "foo", 'c'} # Tuple(Int32, String, Char)
+```
+
+There is also possibility to create a tuple using the `Tuple` class.
+
+```crystal
+tuple = Tuple(Int32, String, Char).new(1, "foo", 'c')
+```
+
+Alternatively, you can explicitly specify the type of the variable assigned to the tuple.
+
+```crystal
+tuple : Tuple(Int32, String, Char) = {1, "foo", 'c'}
+```
+
+Explicitly specifying the type of the tuple can be usefull since that allows for defining that a position should hold a union type.
+This means that a position can hold multiple types.
+
+```crystal
+tuple : Tuple(Int32 | String, String, Char) = {1, "foo", 'c'}
+```
+
+## Conversion
+
+### Creating a tuple from an array
+
+You can create a tuple from an array using the `Tuple` class's `from` method.
+This requires that the type of the tuple is specified.
+
+```crystal
+array = [1, "foo", 'c']
+tuple = Tuple(Int32, String, Char).from(array)
+```
+
+### Conversion to Array
+
+You can convert a tuple to an array using the `to_a` method.
+
+```crystal
+tuple = {1, "foo", 'c'}
+array = tuple.to_a
+array # => [1, "foo", 'c']
+```
+
+### Tuples definition shorthand
+
+There is a shorthand for defining explicit tuple types.
+That is by writting the types inside of the tuple literal.
+
+```crystal
+tuple : {Int32, String, Char} = {1, "foo", 'c'}
```

Comment:
This looks like the normal form for writing an explicit tuple type. Did you intend something like this?

```suggestion
tuple = {1 : Int32, "foo" : String, 'c' : Char}
```

Reply:
In normal cases, you would define the type as `:Tuple(Int32, String, Char)`

Reply:
Ah, my bad. I was hoping there was an even better short form.